snapcraft/_options.py

```diff
@@ -124,7 +124,68 @@ def _get_platform_architecture():
     return architecture
 
 
-class ProjectOptions:
+class ProjectInfo:
+    """Information gained from the snap's snapcraft.yaml file."""
+
+    def __init__(self, *,
+                 name: str, version: str,
+                 summary: str, description: str, confinement: str) -> None:
+        self.__name = name
+        self.__version = version
+        self.__summary = summary
+        self.__description = description
+        self.__confinement = confinement
+
+    @property
+    def name(self):
+        """Returns the name of the snap.
+        """
+        return self.__name
+
+    @property
+    def version(self):
+        """Returns the version of the snap.
+        """
+        return self.__version
+
+    @property
+    def summary(self):
+        """Returns the summary of the snap.
+        """
+        return self.__summary
+
+    @property
+    def description(self):
+        """Returns the description of the snap.
+        """
+        return self.__description
+
+    @property
+    def confinement(self):
+        """Returns the confinement type of the snap.
+        """
+        return self.__confinement
+
+
+class Project:
+    """All details around building a project concerning the build environment
+    and the snap being built."""
+
+    def __init__(self) -> None:
+        self.__info = None
+
+    @property
+    def info(self):
+        """Returns the project info.
+        """
+        return self.__info
+
+    @info.setter
+    def info(self, info):
+        self.__info = info
+
+
+class ProjectOptions(Project):
```

Comment:
How much work would it be to flip this relationship around and replace all the `ProjectOptions` with `Project`s? We're gonna have to do it at some point.

Reply:
That would mean we can never drop `ProjectOptions` or its properties from the API. Right now you have the option of creating a `Project` instance without it.